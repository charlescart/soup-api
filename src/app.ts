/* eslint-disable no-console */
import 'reflect-metadata';
import { useExpressServer } from 'routing-controllers';
import express from 'express';

const app = express();
const port = process.env.PORT || 3000;

useExpressServer(app, {
  routePrefix: '/api',
  cors: true,
  controllers: [`${__dirname}/features/**/*Controller.js`],
});

app.listen(port, () => {
  console.log(`Run in the port ${port}`);
});
