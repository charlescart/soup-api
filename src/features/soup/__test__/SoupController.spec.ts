import app from '../../../server';
import request from 'supertest'

describe('SoupController', () => {
  describe('Sopa de letras irregular y multiples palabras a buscar', () => {
    it('POST /api/soup', done => {
      const searchWords: string[] = ["oiE", "ixE", "Ix", "i", "O", "awds", "e4", "", "oi", "eX"];
      const soup: string[][] = [
        ["o", "I", "E"],
        ["I", "i", "X", "4"],
        ["E", "X", "E"],
        ["E", "X"],
        ["E"],
        [""]
      ];

      request(app).post('/api/soup')
        .send({ soup, searchWords })
        .set('Content-Type', 'application/json')
        .set('Accept', 'application/json')
        .expect(200)
        .end((err, res) => {
          if (err) throw err;
          expect(res.body).toMatchSnapshot();
          done();
        });
    });
  });

  //TODO: agregar los casos de pruebas del documento
});