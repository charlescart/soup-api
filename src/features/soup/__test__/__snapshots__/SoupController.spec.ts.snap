// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoupController Sopa de letras irregular y multiples palabras a buscar POST /api/soup 1`] = `
Object {
  "AWDS": 0,
  "E4": 1,
  "EX": 6,
  "I": 3,
  "IX": 4,
  "IXE": 0,
  "O": 1,
  "OI": 3,
  "OIE": 3,
}
`;
